packages/main/src/TabContainer.ts

```diff
@@ -478,6 +519,148 @@ class TabContainer extends UI5Element {
+			Orientation.Horizontal,
+		);
+		const overflowButton = e.target.closest<HTMLElement>("[data-ui5-stable=overflow-start],[data-ui5-stable=overflow-end]");
+		let popoverTarget = null;
+
+		if (overflowButton) {
+			popoverTarget = overflowButton;
+			e.preventDefault();
+		} else if (closestDropPosition) {
+			const dropTarget = (closestDropPosition.element as Tab).realTabReference;
+			let placements = closestDropPosition.placements;
+
+			if (dropTarget === draggedElement) {
+				placements = placements.filter(placement => placement !== DropPlacement.On);
+			}
+
+			const acceptedPlacement = placements.find(dropPlacement => {
+				const dragOverPrevented = !this.fireEvent<TabContainerMoveOverEventDetail>("move-over", {
+					source: {
+						element: draggedElement!,
+					},
+					destination: {
+						element: dropTarget,
+						placement: dropPlacement,
+					},
+				}, true);
+
+				if (dragOverPrevented) {
+					e.preventDefault();
+					this.dropIndicatorDOM!.targetReference = closestDropPosition.element;
+					this.dropIndicatorDOM!.placement = dropPlacement;
+					return true;
+				}
+
+				return false;
+			});
+
+			if (acceptedPlacement === DropPlacement.On && (closestDropPosition.element as Tab).realTabReference.subTabs.length) {
+				popoverTarget = closestDropPosition.element;
+			}
+		}
+
+		if (popoverTarget && isLongDragOver) {
+			this._showPopoverAt(popoverTarget, false, true);
+		} else {
+			this._closePopover();
+		}
+	}
+
+	_onHeaderDrop(e: DragEvent) {
+		e.preventDefault();
+
+		this.fireEvent<TabContainerMoveEventDetail>("move", {
+			source: {
+				element: DragRegistry.getDraggedElement()!,
+			},
+			destination: {
+				element: (this.dropIndicatorDOM!.targetReference as Tab).realTabReference,
+				placement: this.dropIndicatorDOM!.placement,
+			},
+		});
+
+		this.dropIndicatorDOM!.targetReference = null;
+	}
+
+	_onHeaderDragLeave(e: DragEvent) {
+		if (e.relatedTarget instanceof Node && !this.shadowRoot!.contains(e.relatedTarget)) {
+			this.dropIndicatorDOM!.targetReference = null;
+		}
+	}
+
+	_onHeaderDragEnd() {
+		this.dropIndicatorDOM!.targetReference = null;
+	}
+
+	_onPopoverListMoveOver(e: CustomEvent<ListMoveOverEventDetail>) {
+		const { destination } = e.detail;
+		const draggedElement = DragRegistry.getDraggedElement();
+		const dropTarget = (destination.element as ITab).realTabReference;
+
+		if (destination.placement === DropPlacement.On && (dropTarget.isSeparator || draggedElement === dropTarget)) {
+			return;
+		}
+
+		const placementAccepted = !this.fireEvent<TabContainerMoveOverEventDetail>("move-over", {
+			source: {
+				element: draggedElement!,
+			},
+			destination: {
+				element: dropTarget,
+				placement: destination.placement,
+			},
+		}, true);
+
+		if (placementAccepted) {
+			e.preventDefault();
+		} else {
+			this.dropIndicatorDOM!.targetReference = null;
+		}
+	}
+
+	_onPopoverListMove(e: CustomEvent<ListMoveEventDetail>) {
+		const { destination } = e.detail;
```

Comment:
there is a rare bug (low prio):
- you select a tab
- drag the selected tab in the overflow
- drop it there
- the selected tab is now in the overflow, but since the selected tab must be visible in the header - it automatically pops out of the overflow
- but then the popover still stays open, while it will be better to close

Reply:
I couldn't find solution for this problem, because: 1) we need to keep the popover open, in order to allow for rearranging its items and 2) we need to close the popover, if the selected item is put inside it. Both things should happen `onAfterRendering`.
The only solution I could think of is to disable dropping the selected item inside the popover. I have sent a question to the UX designer.